File: userapi/src/main/java/com/teleri/userapi/model/Filter.java
package com.teleri.userapi.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Filter {
    private String name;
    private String lastname;
    private String email;
    private String nationality;
    private Integer page;
    private Integer size;
}

File: userfront/src/app/models/filter.ts
export class Filter {
    name: string;
    lastname: string;
    email: string;
    nationality: string;

    constructor(name: string, email: string, lastname: string, nationality: string) {
        this.name = name;
        this.email = email;
        this.lastname = lastname;
        this.nationality = nationality;
    }
}
